packages/ui5-task-compileless/lib/compileLess.js

```diff
@@ -0,0 +1,105 @@
+const log = require("@ui5/logger").getLogger("builder:customtask:compileless");
+const less = require("less-openui5");
+const { ReaderCollectionPrioritized, DuplexCollection, fsInterface } = require("@ui5/fs");
+const Memory = require("@ui5/fs").adapters.Memory
+const FileSystem = require("@ui5/fs").adapters.FileSystem;
+const resourceFactory = require("@ui5/fs/lib/resourceFactory");
+const minimatch = require("minimatch");
+
+/**
+ * Custom task to compile less files in the app folder
+ *
+ * @param {Array} lessResources Array of objects containing the less resource and the corresponding project resource path
+ * @param {fsInterface} fs Node.js styled file system interface
+ * @param {boolean} isDebug Flag indicating debug mode
+ * @returns {Promise<Array>} Promise resolving with the created css resources
+ * @private
+ */
+async function compileLess(lessResources, fs, isDebug) {
+    const lessBuilder = new less.Builder({fs});
+
+    return Promise.all(
+        lessResources.map(lessResource => {
+            const appFolderPath = lessResource.resource.getPath();
+            isDebug && log.info(`Compiling file ${lessResource.projectResourcePath}...`);
+            return lessBuilder
+                .build({
+                    lessInputPath: appFolderPath
+                })
+                .then(output => resourceFactory.createResource({
+                        path: lessResource.projectResourcePath.replace(/(.*).less/, "$1.css"),
+                        string: output.css
+                    })
+                ).catch(error => log.error(error));
+        })
+    );
+}
+
+/**
+ * Custom task to compile less files in the app folder
+ *
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @param {string} [parameters.options.configuration] Task configuration if given in ui5.yaml
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function({workspace, dependencies, options}) {
+    const appFolderPath = options.configuration && options.configuration.appFolderPath || 'webapp';
+
+    const isDebug = options.configuration && options.configuration.debug;
+
+    let manifest = await (await workspace.byPath(`/resources/${options.projectNamespace}/manifest.json`)).getString();
+    manifest = JSON.parse(manifest)
+
+
+    let lessToCompile = options.configuration && options.configuration.lessToCompile|| [] ;
+    lessToCompile = lessToCompile.length ? lessToCompile : manifest["sap.ui5"]?.resources?.css?.map(style => style?.uri?.replace(".css",".less"));
```

Comment:
I guess we can't use optional chaining, as it's not supported in Node versions <= 13.14.0 ([Node Green](https://node.green/))

Reply:
done